### ISSUE TITLE
Second checklist item not cleared when using Voiceover

### ISSUE THREAD

**Issue by CuriousMagpie** [HabitRPG/habitrpg-ios#1225]
**Describe the bug**
The second item on any checklist does not get marked as cleared when using Voiceover, it only activates the daily to be edited. Items after the second in a  checklist work normally when pressed, it is consistently the second item, even in lists of only two. 

**Smartphone (please complete the following information):**
- iOS Version: 15.6
- Device: iPhone14,3
- App Version: 3.5.4 (694)
- User ID: 3eaa4d67-f70c-438c-bdf8-2a2ac926c00c

**Comment by dan-e-miller** [comment]
Hello!

I would like to work on this issue. My Habitica user is @miller4147

Thanks

**Comment by saraolson** [comment]
@dan-e-miller i've marked the issue as in progress for you, let us know if you have to drop it for any reason!

**Comment by dan-e-miller** [comment]
Sounds good @saraolson! Will do - I posted a pull request for the first previous issue I requested to work on :)

**Comment by dan-e-miller** [comment]
Root cause is a clear-colored cell separator that is always above the second checklist item. Since this is a cell subview, the voiceover selection triggers the cell selection action. Working on a fix for this.

**Comment by dan-e-miller** [comment]
Hi @saraolson - I'm running into some snags trying to progress on this issue. The accessibility inspector does not function on the current version of Xcode and I don't have a physical device provisioned, so there's really no way for me to test VoiceOver and validate changes I make.

To pass on what I've discovered to a contributor who may work on this in the future - TaskTableViewCells are, for some reason yet unclear to me, getting created with 2-3 cell separators. All but one are removed if I set `separatorStyle = .none` on the UITableView, but one, specifically the one that sits over the second checklist cell when expanded, persists even when using .none. As mentioned earlier, I believe this separator is what is causing the overall cell to receive the VoiceOver action such that the form is opened instead of the checklist item being completed.

<img width="955" alt="screenshot" src="https://user-images.githubusercontent.com/81938951/187928652-68d11943-797b-492a-b2ec-78e49a18c925.png">

I would be happy to resume work on this issue in the future if Accessibility Inspector functionality is restored or if it would be possible to get a device provisioned.

**Comment by saraolson** [comment]
@dan-e-miller thanks for letting us know and making this much progress! i'll remove the in-progress label. our developer may have some time to pick this up and carry it through from where you left off

**Comment by AriPearce** [comment]
Hi! I would like to work on this issue (I have a physical device to test on so I should be able to pick up where Dan left off). My Habitica user is @aripepear.

Thanks!

**Comment by saraolson** [comment]
@AriPearce sorry for the delay! if you're still interested in taking this on you're welcome to.

**Comment by AriPearce** [comment]
@saraolson Happy to look into it!

**Comment by AriPearce** [comment]
Hi @saraolson, unfortunately I'm unable to replicate this issue on my device (iPhone 11, iOS 16.3.1, app version 3.8.1) so I won't be able to help much. Hopefully someone else is able to make progress!

**Comment by saraolson** [comment]
Testing while working on other voiceover issues, i also wasnt able to replicate so i'm going to close this issue. if someone else encounters it again, we can reopen.

**Comment by rilygreat** [comment]
> iOS Version: 18.6.2
> Device: iPhone17,3
> App Version: 3.14.4 (1014)
> 
> Please describe the bug you encountered:
> 
> When using Voiceover with the iPhone app, for tasks with checklists, if you try to double tap on the second checklist item, it does not check the box, but instead opens up the task to edit it. This seems to be exclusively an issue with the second checklist item, and I have noticed this with both dailies and to do’s. Thank you!
